### PR TITLE
fix(files): Add more visual move / copy notification

### LINKED ISSUE #46645: [Bug]: No visual feedback for move operations
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Ther's no visual feedback for a move operation.

### Steps to reproduce

1. Add an external storage
1. Have a large file (e.g. an ubuntu desktop iso)
2. Move the large file from your home folder to the external storage
3. There's no feedback that anything is happening

Nextcloud 28:

[9664474 28.webm](https://github.com/user-attachments/assets/77329954-783d-4f9c-8317-0294e697ad14)

Nextcloud 27:

[9664474 27.webm](https://github.com/user-attachments/assets/98963bae-4e0d-4ead-b75d-818746fad359)

### Expected behavior

Feedback that something is happening. 

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

I'm seeing a progress indicator with Nextcloud 30 again.

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: https://github.com/nextcloud/server/issues/46645

## Summary

This adds loading toast notification while the move operation is running.

## Screen recording

[Bildschirmaufnahme_20240911_222420.webm](https://github.com/user-attachments/assets/e57cc014-ed75-4628-b9e0-2cf5c35cde18)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by kesselb** [comment]
image

Somehow I have less spacing between the icon and text with my Firefox :confused: 

Is it on purpose that the toast is only shown when using the filepicker, but not when using drag and drop?

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php [hunk 1]
```diff
@@ -187,6 +187,7 @@ public function checkCopy($source, $target): void {
 	 * @throws InvalidPath If the target name is invalid
 	 */
 	public function checkMove(string $source, string $target): void {
+		sleep(10);
```
Comment (kesselb) [inline apps/dav/lib/Connector/Sabre/FilesPlugin.php:190]: Handy for testing, please don't forget to remove it later ;)

Comment (susnux) [reply apps/dav/lib/Connector/Sabre/FilesPlugin.php:190]: ⚠️⚠️⚠️

**Review by marcoambrosini — COMMENTED** [review]
Very nice, can we use a loader component instead and feed the actual progress of the operation to it? 

I had [a mockup for something similar](https://github.com/nextcloud/server/issues/41790) a few months ago, but this is a much simpler solution that does a big part of the job.

**Comment by skjnldsv** [comment]
> Very nice, can we use a loader component instead and feed the actual progress of the operation to it?

That is much more work, and should be scheduled by a team lead :)

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php (hunk 1 above)
Comment (susnux) [reply apps/dav/lib/Connector/Sabre/FilesPlugin.php:190]:
```suggestion
```

File: apps/files/css/files.scss [hunk 2]
```diff
@@ -6,6 +6,13 @@
 @use 'variables';
 @import 'functions';
 
+.toast-loading-icon {
+	// Reduce start margin (it was made for text but this is an icon)
+	margin-inline-start: -8px;
```
Comment (susnux) [inline apps/files/css/files.scss:11]:
```suggestion
	margin-inline-start: -4px;
```

**Review by kesselb — APPROVED** [review]

**Comment by kesselb** [comment]
Thank you :+1: 

It's not ideal that for every selected file an toast is shown, but still better than before :pray:

**Review by marcoambrosini — APPROVED** [review]

**Comment by kesselb** [comment]
Rebase + Recompile

**Pull request merged**